MatrixSDKTests/MXRoomStateDynamicTests.m

```diff
@@ -203,73 +205,76 @@ - (void)testLiveEventsForScenario1
             MXRoom *room = [mxSession roomWithRoomId:roomId];
             
             __block NSUInteger eventCount = 0;
-            [room.liveTimeline listenToEventsOfTypes:nil onEvent:^(MXEvent *event, MXTimelineDirection direction, MXRoomState *roomState) {
-                
-                // Check each expected event and their roomState contect
-                // Events are live. Then comes in order
-                switch (eventCount++) {
+            [room liveTimeline:^(MXEventTimeline *liveTimeline) {
+                [liveTimeline listenToEventsOfTypes:nil onEvent:^(MXEvent *event, MXTimelineDirection direction, MXRoomState *roomState) {
 
-                    case 0:
-                        //  2 - Bob: "Hello World"
-                        XCTAssertEqual(event.eventType, MXEventTypeRoomMessage);
-                        
-                        XCTAssertNotNil(roomState);
-                        
-                        XCTAssertNil(roomState.topic, @"The room topic is not yet defined. Found: %@", roomState.topic);
-                        XCTAssertNil(room.state.topic, @"The room topic is not yet defined. Found: %@", roomState.topic);
-                        break;
-                        
-                    case 1:
-                        //  3 - Bob changes the room topic to "Topic #1"
-                        XCTAssertEqual(event.eventType, MXEventTypeRoomTopic);
+                    // Check each expected event and their roomState contect
+                    // Events are live. Then comes in order
+                    switch (eventCount++) {
```

Comment:
Maybe more readable to switch on `eventCount` and then call `eventCount++` after switching

Reply:
I guess that depends on your C background :)